File: Samples/Silverlight_VB/EntitySpaces.Proxies.Silverlight/Proxies/Employees.vb
'===============================================================================
'                   EntitySpaces Studio by EntitySpaces, LLC
'            Persistence Layer and Business Objects for Microsoft .NET
'            EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
'                         http://www.entityspaces.net
'===============================================================================
' EntitySpaces Version : 2012.1.0930.0
' EntitySpaces Driver  : SQL
' Date Generated       : 9/23/2012 6:16:24 PM
'===============================================================================

Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Text
Imports System.Xml.Serialization
Imports System.Runtime.Serialization

Imports EntitySpaces.DynamicQuery

Namespace Proxies

	<DataContract(Namespace:="http://tempuri.org/", Name:="EmployeesCollection")> _
	<XmlType(TypeName:="EmployeesCollectionProxyStub")> _	
	Partial Public Class EmployeesCollectionProxyStub 

		Public Sub New() 
		
		End Sub
		
		<DataMember(Name:="Collection", EmitDefaultValue:=False)> _
		Public Collection As New ObservableCollection (Of EmployeesProxyStub)
		
		Public Function IsDirty() As Boolean
			For Each obj As EmployeesProxyStub In Collection
				If obj.IsDirty() Then
					Return True
				End If
			Next
			Return False
		End Function
	

	End Class
	
	<DataContract(Namespace:="http://tempuri.org/", Name:="Employees")> _
	<XmlType(TypeName:="EmployeesProxyStub")> _	
	Partial Public Class EmployeesProxyStub
		Implements System.ComponentModel.INotifyPropertyChanged
		
		Public Sub New()
			Me.esRowState = "Added"
		End Sub
		
		Public Function IsDirty() As Boolean
			Return If(esRowState <> "Unchanged", True, False)
		End Function
		
		Public Event PropertyChanged As System.ComponentModel.PropertyChangedEventHandler Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
		
		Protected Sub RaisePropertyChanged(ByVal propertyName As String)
			Dim propertyChanged As System.ComponentModel.PropertyChangedEventHandler = Me.PropertyChangedEvent
			If (Not (propertyChanged) Is Nothing) Then
				propertyChanged(Me, New System.ComponentModel.PropertyChangedEventArgs(propertyName))
			End If
		End Sub
		
		<DataMember(Name:="a0", Order:=1, EmitDefaultValue:=False)> _
        Public Property EmployeeID As Nullable(Of System.Int32)
			Get
				Return _EmployeeID
			End Get
			Set(ByVal value As Nullable(Of System.Int32))
				If Not _EmployeeID.Equals(value) Then
					_EmployeeID = value
					SetDirty("EmployeeID")
					RaisePropertyChanged("EmployeeID")
				End If
			End Set
		End Property
		Private _EmployeeID As Nullable(Of System.Int32)
		<DataMember(Name:="a1", Order:=2, EmitDefaultValue:=False)> _
        Public Property LastName As System.String
			Get
				Return _LastName
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_LastName, value) = 0 Then
					_LastName = value
					SetDirty("LastName")
					RaisePropertyChanged("LastName")
				End If
			End Set
		End Property
		Private _LastName As System.String
		<DataMember(Name:="a2", Order:=3, EmitDefaultValue:=False)> _
        Public Property FirstName As System.String
			Get
				Return _FirstName
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_FirstName, value) = 0 Then
					_FirstName = value
					SetDirty("FirstName")
					RaisePropertyChanged("FirstName")
				End If
			End Set
		End Property
		Private _FirstName As System.String
		<DataMember(Name:="a3", Order:=4, EmitDefaultValue:=False)> _
        Public Property Title As System.String
			Get
				Return _Title
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Title, value) = 0 Then
					_Title = value
					SetDirty("Title")
					RaisePropertyChanged("Title")
				End If
			End Set
		End Property
		Private _Title As System.String
		<DataMember(Name:="a4", Order:=5, EmitDefaultValue:=False)> _
        Public Property TitleOfCourtesy As System.String
			Get
				Return _TitleOfCourtesy
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_TitleOfCourtesy, value) = 0 Then
					_TitleOfCourtesy = value
					SetDirty("TitleOfCourtesy")
					RaisePropertyChanged("TitleOfCourtesy")
				End If
			End Set
		End Property
		Private _TitleOfCourtesy As System.String
		<DataMember(Name:="a5", Order:=6, EmitDefaultValue:=False)> _
        Public Property BirthDate As Nullable(Of System.DateTime)
			Get
				Return _BirthDate
			End Get
			Set(ByVal value As Nullable(Of System.DateTime))
				If Not _BirthDate.Equals(value) Then
					_BirthDate = value
					SetDirty("BirthDate")
					RaisePropertyChanged("BirthDate")
				End If
			End Set
		End Property
		Private _BirthDate As Nullable(Of System.DateTime)
		<DataMember(Name:="a6", Order:=7, EmitDefaultValue:=False)> _
        Public Property HireDate As Nullable(Of System.DateTime)
			Get
				Return _HireDate
			End Get
			Set(ByVal value As Nullable(Of System.DateTime))
				If Not _HireDate.Equals(value) Then
					_HireDate = value
					SetDirty("HireDate")
					RaisePropertyChanged("HireDate")
				End If
			End Set
		End Property
		Private _HireDate As Nullable(Of System.DateTime)
		<DataMember(Name:="a7", Order:=8, EmitDefaultValue:=False)> _
        Public Property Address As System.String
			Get
				Return _Address
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Address, value) = 0 Then
					_Address = value
					SetDirty("Address")
					RaisePropertyChanged("Address")
				End If
			End Set
		End Property
		Private _Address As System.String
		<DataMember(Name:="a8", Order:=9, EmitDefaultValue:=False)> _
        Public Property City As System.String
			Get
				Return _City
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_City, value) = 0 Then
					_City = value
					SetDirty("City")
					RaisePropertyChanged("City")
				End If
			End Set
		End Property
		Private _City As System.String
		<DataMember(Name:="a9", Order:=10, EmitDefaultValue:=False)> _
        Public Property Region As System.String
			Get
				Return _Region
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Region, value) = 0 Then
					_Region = value
					SetDirty("Region")
					RaisePropertyChanged("Region")
				End If
			End Set
		End Property
		Private _Region As System.String
		<DataMember(Name:="a10", Order:=11, EmitDefaultValue:=False)> _
        Public Property PostalCode As System.String
			Get
				Return _PostalCode
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_PostalCode, value) = 0 Then
					_PostalCode = value
					SetDirty("PostalCode")
					RaisePropertyChanged("PostalCode")
				End If
			End Set
		End Property
		Private _PostalCode As System.String
		<DataMember(Name:="a11", Order:=12, EmitDefaultValue:=False)> _
        Public Property Country As System.String
			Get
				Return _Country
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Country, value) = 0 Then
					_Country = value
					SetDirty("Country")
					RaisePropertyChanged("Country")
				End If
			End Set
		End Property
		Private _Country As System.String
		<DataMember(Name:="a12", Order:=13, EmitDefaultValue:=False)> _
        Public Property HomePhone As System.String
			Get
				Return _HomePhone
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_HomePhone, value) = 0 Then
					_HomePhone = value
					SetDirty("HomePhone")
					RaisePropertyChanged("HomePhone")
				End If
			End Set
		End Property
		Private _HomePhone As System.String
		<DataMember(Name:="a13", Order:=14, EmitDefaultValue:=False)> _
        Public Property Extension As System.String
			Get
				Return _Extension
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Extension, value) = 0 Then
					_Extension = value
					SetDirty("Extension")
					RaisePropertyChanged("Extension")
				End If
			End Set
		End Property
		Private _Extension As System.String
		<DataMember(Name:="a14", Order:=15, EmitDefaultValue:=False)> _
        Public Property Photo As System.Byte()
			Get
				Return _Photo
			End Get
			Set(ByVal value As System.Byte())
				If _Photo Is value Then
					_Photo = value
					SetDirty("Photo")
					RaisePropertyChanged("Photo")
				End If
			End Set
		End Property
		Private _Photo As System.Byte()
		<DataMember(Name:="a15", Order:=16, EmitDefaultValue:=False)> _
        Public Property Notes As System.String
			Get
				Return _Notes
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Notes, value) = 0 Then
					_Notes = value
					SetDirty("Notes")
					RaisePropertyChanged("Notes")
				End If
			End Set
		End Property
		Private _Notes As System.String
		<DataMember(Name:="a16", Order:=17, EmitDefaultValue:=False)> _
        Public Property ReportsTo As Nullable(Of System.Int32)
			Get
				Return _ReportsTo
			End Get
			Set(ByVal value As Nullable(Of System.Int32))
				If Not _ReportsTo.Equals(value) Then
					_ReportsTo = value
					SetDirty("ReportsTo")
					RaisePropertyChanged("ReportsTo")
				End If
			End Set
		End Property
		Private _ReportsTo As Nullable(Of System.Int32)
		<DataMember(Name:="a17", Order:=18, EmitDefaultValue:=False)> _
        Public Property PhotoPath As System.String
			Get
				Return _PhotoPath
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_PhotoPath, value) = 0 Then
					_PhotoPath = value
					SetDirty("PhotoPath")
					RaisePropertyChanged("PhotoPath")
				End If
			End Set
		End Property
		Private _PhotoPath As System.String
		
		

		<DataMember(Name:="a10000", Order:=10000)> _
		Public Property esRowState As String
			Get
				Return Me._esRowState
			End Get
			Set
				If Not String.Compare(_esRowState, value) = 0 Then
					Me._esRowState = value
					RaisePropertyChanged("esRowState")
				End If
			End Set
		End Property
		Private _esRowState As String = "Unchanged"
		
		<DataMember(Name := "a10001", Order := 10001, EmitDefaultValue := False)> _		
		Public Property ModifiedColumns() As List(Of String)
			Get
				If _ModifiedColumns Is Nothing Then
					_ModifiedColumns = New List(Of String)()
				End If
				Return _ModifiedColumns
			End Get
			Set(ByVal value As List(Of String))
				_ModifiedColumns = New List(Of String)(value)
			End Set
		End Property
		Private _ModifiedColumns As List(Of String)

		<DataMember(Name := "a10002", Order := 10002, EmitDefaultValue := False)> _
		Public Property esExtraColumns() As Dictionary(Of String, Object)
			Get
				If _ExtraColumns Is Nothing Then
					_ExtraColumns = New Dictionary(Of String, Object)()
				End If
				Return _ExtraColumns
			End Get
			Set(ByVal value As Dictionary(Of String, Object))
				_ExtraColumns = New Dictionary(Of String, Object)(value)
			End Set
		End Property
		Private _ExtraColumns As Dictionary(Of String, Object)

		Public Sub MarkAsDeleted()
			Me.esRowState = "Deleted"
		End Sub

		Private Sub SetDirty(ByVal [property] As String)
			If Not ModifiedColumns.Contains([property]) Then
				ModifiedColumns.Add([property])
			End If
			
			If Me.esRowState = "Unchanged" Then
				Me.esRowState = "Modified"
			End If
		End Sub
		
	End Class
	
	<XmlType(TypeName:="EmployeesQueryProxyStub")> _	
	<DataContract(Name:="EmployeesQuery", Namespace:= "http://www.entityspaces.net")> _	
	Partial Public Class EmployeesQueryProxyStub 
		Inherits esDynamicQuerySerializable

		Public Sub New()
		
		End Sub

		Public Sub New(ByVal joinAlias As String)
			MyBase.es.JoinAlias = joinAlias
		End Sub	
		
		
		Protected Overrides Function GetQueryName() As String
			Return "EmployeesQuery"
		End Function

		#Region "Implicit Casts"

		Public Shared Widening Operator CType(ByVal query As EmployeesQueryProxyStub) As String
			Return EmployeesQueryProxyStub.SerializeHelper.ToXml(query)
		End Operator

		#End Region
		

		Public ReadOnly Property EmployeeID As esQueryItem
			Get
				Return new esQueryItem(Me, "EmployeeID", esSystemType.Int32)
			End Get
		End Property 
		
		Public ReadOnly Property LastName As esQueryItem
			Get
				Return new esQueryItem(Me, "LastName", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property FirstName As esQueryItem
			Get
				Return new esQueryItem(Me, "FirstName", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Title As esQueryItem
			Get
				Return new esQueryItem(Me, "Title", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property TitleOfCourtesy As esQueryItem
			Get
				Return new esQueryItem(Me, "TitleOfCourtesy", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property BirthDate As esQueryItem
			Get
				Return new esQueryItem(Me, "BirthDate", esSystemType.DateTime)
			End Get
		End Property 
		
		Public ReadOnly Property HireDate As esQueryItem
			Get
				Return new esQueryItem(Me, "HireDate", esSystemType.DateTime)
			End Get
		End Property 
		
		Public ReadOnly Property Address As esQueryItem
			Get
				Return new esQueryItem(Me, "Address", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property City As esQueryItem
			Get
				Return new esQueryItem(Me, "City", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Region As esQueryItem
			Get
				Return new esQueryItem(Me, "Region", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property PostalCode As esQueryItem
			Get
				Return new esQueryItem(Me, "PostalCode", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Country As esQueryItem
			Get
				Return new esQueryItem(Me, "Country", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property HomePhone As esQueryItem
			Get
				Return new esQueryItem(Me, "HomePhone", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Extension As esQueryItem
			Get
				Return new esQueryItem(Me, "Extension", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Photo As esQueryItem
			Get
				Return new esQueryItem(Me, "Photo", esSystemType.ByteArray)
			End Get
		End Property 
		
		Public ReadOnly Property Notes As esQueryItem
			Get
				Return new esQueryItem(Me, "Notes", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property ReportsTo As esQueryItem
			Get
				Return new esQueryItem(Me, "ReportsTo", esSystemType.Int32)
			End Get
		End Property 
		
		Public ReadOnly Property PhotoPath As esQueryItem
			Get
				Return new esQueryItem(Me, "PhotoPath", esSystemType.String)
			End Get
		End Property 
		
	
	End Class
	
End Namespace

File: Samples/WindowsPhone/EntitySpaces.Proxies/ProxyStub/Employees.cs
/*
===============================================================================
                    EntitySpaces Studio by EntitySpaces, LLC
             Persistence Layer and Business Objects for Microsoft .NET
             EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
                          http://www.entityspaces.net
===============================================================================
EntitySpaces Version : 2012.1.0930.0
EntitySpaces Driver  : SQL
Date Generated       : 9/23/2012 6:16:28 PM
===============================================================================
*/

using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Text;
using System.Xml.Serialization;
using System.Runtime.Serialization;

using EntitySpaces.DynamicQuery;

namespace Proxies
{
	[DataContract(Namespace = "http://tempuri.org/", Name = "EmployeesCollection")]	
	[XmlType(TypeName = "EmployeesCollectionProxyStub")]	
	public partial class EmployeesCollectionProxyStub 
	{
		public EmployeesCollectionProxyStub() { }
	
		[DataMember(Name = "Collection", EmitDefaultValue = false)]
		public ObservableCollection<EmployeesProxyStub> Collection = new ObservableCollection<EmployeesProxyStub>();
		
		public bool IsDirty()
		{
			foreach (EmployeesProxyStub obj in Collection)
			{
				if (obj.IsDirty()) return true;
			}

			return false;
		}
	}
	
	[DataContract(Namespace = "http://tempuri.org/", Name = "Employees")]
	[XmlType(TypeName = "EmployeesProxyStub")]	
	public partial class EmployeesProxyStub : object, System.ComponentModel.INotifyPropertyChanged
	{
		public EmployeesProxyStub()
		{
			this.esRowState = "Added";
		}
		
		public bool IsDirty()
		{
			return esRowState != "Unchanged" ? true : false;
		}
		
		public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
		
		protected void RaisePropertyChanged(string propertyName) 
		{
			System.ComponentModel.PropertyChangedEventHandler propertyChanged = this.PropertyChanged;
			if ((propertyChanged != null)) 
			{
				propertyChanged(this, new System.ComponentModel.PropertyChangedEventArgs(propertyName));
			}
		}
		
		[DataMember(Name="a0", Order=1, EmitDefaultValue=false)]
		public System.Int32? EmployeeID
		{
			get
			{
				return _EmployeeID;
			}
			set
			{
				if (_EmployeeID != value)
				{
					_EmployeeID = value;
					SetDirty("EmployeeID");
					RaisePropertyChanged("EmployeeID");
				}
			}
		}
		private System.Int32? _EmployeeID;
		
		[DataMember(Name="a1", Order=2, EmitDefaultValue=false)]
		public System.String LastName
		{
			get
			{
				return _LastName;
			}
			set
			{
				if (_LastName != value)
				{
					_LastName = value;
					SetDirty("LastName");
					RaisePropertyChanged("LastName");
				}
			}
		}
		private System.String _LastName;
		
		[DataMember(Name="a2", Order=3, EmitDefaultValue=false)]
		public System.String FirstName
		{
			get
			{
				return _FirstName;
			}
			set
			{
				if (_FirstName != value)
				{
					_FirstName = value;
					SetDirty("FirstName");
					RaisePropertyChanged("FirstName");
				}
			}
		}
		private System.String _FirstName;
		
		[DataMember(Name="a3", Order=4, EmitDefaultValue=false)]
		public System.String Title
		{
			get
			{
				return _Title;
			}
			set
			{
				if (_Title != value)
				{
					_Title = value;
					SetDirty("Title");
					RaisePropertyChanged("Title");
				}
			}
		}
		private System.String _Title;
		
		[DataMember(Name="a4", Order=5, EmitDefaultValue=false)]
		public System.String TitleOfCourtesy
		{
			get
			{
				return _TitleOfCourtesy;
			}
			set
			{
				if (_TitleOfCourtesy != value)
				{
					_TitleOfCourtesy = value;
					SetDirty("TitleOfCourtesy");
					RaisePropertyChanged("TitleOfCourtesy");
				}
			}
		}
		private System.String _TitleOfCourtesy;
		
		[DataMember(Name="a5", Order=6, EmitDefaultValue=false)]
		public System.DateTime? BirthDate
		{
			get
			{
				return _BirthDate;
			}
			set
			{
				if (_BirthDate != value)
				{
					_BirthDate = value;
					SetDirty("BirthDate");
					RaisePropertyChanged("BirthDate");
				}
			}
		}
		private System.DateTime? _BirthDate;
		
		[DataMember(Name="a6", Order=7, EmitDefaultValue=false)]
		public System.DateTime? HireDate
		{
			get
			{
				return _HireDate;
			}
			set
			{
				if (_HireDate != value)
				{
					_HireDate = value;
					SetDirty("HireDate");
					RaisePropertyChanged("HireDate");
				}
			}
		}
		private System.DateTime? _HireDate;
		
		[DataMember(Name="a7", Order=8, EmitDefaultValue=false)]
		public System.String Address
		{
			get
			{
				return _Address;
			}
			set
			{
				if (_Address != value)
				{
					_Address = value;
					SetDirty("Address");
					RaisePropertyChanged("Address");
				}
			}
		}
		private System.String _Address;
		
		[DataMember(Name="a8", Order=9, EmitDefaultValue=false)]
		public System.String City
		{
			get
			{
				return _City;
			}
			set
			{
				if (_City != value)
				{
					_City = value;
					SetDirty("City");
					RaisePropertyChanged("City");
				}
			}
		}
		private System.String _City;
		
		[DataMember(Name="a9", Order=10, EmitDefaultValue=false)]
		public System.String Region
		{
			get
			{
				return _Region;
			}
			set
			{
				if (_Region != value)
				{
					_Region = value;
					SetDirty("Region");
					RaisePropertyChanged("Region");
				}
			}
		}
		private System.String _Region;
		
		[DataMember(Name="a10", Order=11, EmitDefaultValue=false)]
		public System.String PostalCode
		{
			get
			{
				return _PostalCode;
			}
			set
			{
				if (_PostalCode != value)
				{
					_PostalCode = value;
					SetDirty("PostalCode");
					RaisePropertyChanged("PostalCode");
				}
			}
		}
		private System.String _PostalCode;
		
		[DataMember(Name="a11", Order=12, EmitDefaultValue=false)]
		public System.String Country
		{
			get
			{
				return _Country;
			}
			set
			{
				if (_Country != value)
				{
					_Country = value;
					SetDirty("Country");
					RaisePropertyChanged("Country");
				}
			}
		}
		private System.String _Country;
		
		[DataMember(Name="a12", Order=13, EmitDefaultValue=false)]
		public System.String HomePhone
		{
			get
			{
				return _HomePhone;
			}
			set
			{
				if (_HomePhone != value)
				{
					_HomePhone = value;
					SetDirty("HomePhone");
					RaisePropertyChanged("HomePhone");
				}
			}
		}
		private System.String _HomePhone;
		
		[DataMember(Name="a13", Order=14, EmitDefaultValue=false)]
		public System.String Extension
		{
			get
			{
				return _Extension;
			}
			set
			{
				if (_Extension != value)
				{
					_Extension = value;
					SetDirty("Extension");
					RaisePropertyChanged("Extension");
				}
			}
		}
		private System.String _Extension;
		
		[DataMember(Name="a14", Order=15, EmitDefaultValue=false)]
		public System.Byte[] Photo
		{
			get
			{
				return _Photo;
			}
			set
			{
				if (_Photo != value)
				{
					_Photo = value;
					SetDirty("Photo");
					RaisePropertyChanged("Photo");
				}
			}
		}
		private System.Byte[] _Photo;
		
		[DataMember(Name="a15", Order=16, EmitDefaultValue=false)]
		public System.String Notes
		{
			get
			{
				return _Notes;
			}
			set
			{
				if (_Notes != value)
				{
					_Notes = value;
					SetDirty("Notes");
					RaisePropertyChanged("Notes");
				}
			}
		}
		private System.String _Notes;
		
		[DataMember(Name="a16", Order=17, EmitDefaultValue=false)]
		public System.Int32? ReportsTo
		{
			get
			{
				return _ReportsTo;
			}
			set
			{
				if (_ReportsTo != value)
				{
					_ReportsTo = value;
					SetDirty("ReportsTo");
					RaisePropertyChanged("ReportsTo");
				}
			}
		}
		private System.Int32? _ReportsTo;
		
		[DataMember(Name="a17", Order=18, EmitDefaultValue=false)]
		public System.String PhotoPath
		{
			get
			{
				return _PhotoPath;
			}
			set
			{
				if (_PhotoPath != value)
				{
					_PhotoPath = value;
					SetDirty("PhotoPath");
					RaisePropertyChanged("PhotoPath");
				}
			}
		}
		private System.String _PhotoPath;
		
	
		[DataMember(Name="a10000", Order=10000)]
		public string esRowState
		{
			get { return _esRowState; }
			set 
			{
				if(_esRowState != value)
				{
					_esRowState = value; 
					RaisePropertyChanged("esRowState");
				}
			}
		}
		private string _esRowState = "Unchanged";
		
		[DataMember(Name="a10001", Order=10001, EmitDefaultValue=false)]
		public List<string> ModifiedColumns
		{
			get
			{
				if (_ModifiedColumns == null)
				{
					_ModifiedColumns = new List<string>();
				}
				return _ModifiedColumns;
			}
			set
			{
				_ModifiedColumns = new List<string>(value);
			}
		}
		List<string> _ModifiedColumns;
		
		[DataMember(Name="a10002", Order=10002, EmitDefaultValue=false)]		
		public Dictionary<string, object> esExtraColumns
		{
			get
			{
				if (_ExtraColumns == null)
				{
					_ExtraColumns = new Dictionary<string, object>();
				}
				return _ExtraColumns;
			}
			set
			{
				_ExtraColumns = new Dictionary<string, object>(value);
			}
		}
		Dictionary<string, object> _ExtraColumns;

		public void MarkAsDeleted()
		{
			this.esRowState = "Deleted";
		}

		private void SetDirty(string property)
		{
			if (!ModifiedColumns.Contains(property))
			{
				ModifiedColumns.Add(property);
			}

			if (this.esRowState == "Unchanged")
			{
				this.esRowState = "Modified";
			}
		}
	}
	
	[XmlType(TypeName = "EmployeesQueryProxyStub")]	
	[DataContract(Name = "EmployeesQuery", Namespace = "http://www.entityspaces.net")]
	public partial class EmployeesQueryProxyStub : esDynamicQuerySerializable
	{
		public EmployeesQueryProxyStub() { }
		
		public EmployeesQueryProxyStub(string joinAlias)
		{
			this.es.JoinAlias = joinAlias;
		}	
		
		override protected string GetQueryName()
		{
			return "EmployeesQuery";
		}
		
		#region Implicit Casts

		public static implicit operator string(EmployeesQueryProxyStub query)
		{
			return Proxies.EmployeesQueryProxyStub.SerializeHelper.ToXml(query);
		}

		#endregion
	
		public esQueryItem EmployeeID
		{
			get { return new esQueryItem(this, "EmployeeID", esSystemType.Int32); }
		} 
		
		public esQueryItem LastName
		{
			get { return new esQueryItem(this, "LastName", esSystemType.String); }
		} 
		
		public esQueryItem FirstName
		{
			get { return new esQueryItem(this, "FirstName", esSystemType.String); }
		} 
		
		public esQueryItem Title
		{
			get { return new esQueryItem(this, "Title", esSystemType.String); }
		} 
		
		public esQueryItem TitleOfCourtesy
		{
			get { return new esQueryItem(this, "TitleOfCourtesy", esSystemType.String); }
		} 
		
		public esQueryItem BirthDate
		{
			get { return new esQueryItem(this, "BirthDate", esSystemType.DateTime); }
		} 
		
		public esQueryItem HireDate
		{
			get { return new esQueryItem(this, "HireDate", esSystemType.DateTime); }
		} 
		
		public esQueryItem Address
		{
			get { return new esQueryItem(this, "Address", esSystemType.String); }
		} 
		
		public esQueryItem City
		{
			get { return new esQueryItem(this, "City", esSystemType.String); }
		} 
		
		public esQueryItem Region
		{
			get { return new esQueryItem(this, "Region", esSystemType.String); }
		} 
		
		public esQueryItem PostalCode
		{
			get { return new esQueryItem(this, "PostalCode", esSystemType.String); }
		} 
		
		public esQueryItem Country
		{
			get { return new esQueryItem(this, "Country", esSystemType.String); }
		} 
		
		public esQueryItem HomePhone
		{
			get { return new esQueryItem(this, "HomePhone", esSystemType.String); }
		} 
		
		public esQueryItem Extension
		{
			get { return new esQueryItem(this, "Extension", esSystemType.String); }
		} 
		
		public esQueryItem Photo
		{
			get { return new esQueryItem(this, "Photo", esSystemType.ByteArray); }
		} 
		
		public esQueryItem Notes
		{
			get { return new esQueryItem(this, "Notes", esSystemType.String); }
		} 
		
		public esQueryItem ReportsTo
		{
			get { return new esQueryItem(this, "ReportsTo", esSystemType.Int32); }
		} 
		
		public esQueryItem PhotoPath
		{
			get { return new esQueryItem(this, "PhotoPath", esSystemType.String); }
		} 
		
	}
}